osm2streets/src/lib.rs

```diff
@@ -544,6 +573,11 @@ pub struct Intersection {
     pub control: ControlType,
     pub elevation: Distance,
 
+    /// All roads connected to this intersection. They may be incoming or outgoing relative to this
+    /// intersection.
+    /// TODO: Guarantee they're sorted clockwise
+    pub roads: Vec<OriginalRoad>,
```

Comment:
Is there a good type that has an `insert_at` to take advantage of the fact that roads is already sorted?

Reply:
Like https://doc.rust-lang.org/std/vec/struct.Vec.html#method.insert ? When we're adding a new road, we could calculate its position in the list and add it, or just recalculate the ordering for the intersection. It'll depend on the sorting algorithm

Reply:
Right. Performance doesn't really matter, so `Vec` works if we want to do it the "find and insert" way.